core/api/src/main/java/com/microsoft/gctoolkit/event/g1gc/G1Young.java

```diff
@@ -213,9 +215,9 @@ public void tableFixupStatistics(StatisticalSummary summary) {
         this.tableFixupStatistics = summary;
     }
 
-    private HashMap<String, Double> preEvacuateCSetPhase = new HashMap<>(3);
-    private HashMap<String, UnifiedStatisticalSummary> evacuateCSetPhase = new HashMap<>();
-    private HashMap<String, Double> postEvacuateCSetPhase = new HashMap<>();
+    private final Map<String, Double> preEvacuateCSetPhase = new ConcurrentHashMap<>(3);
```

Comment:
What is the thinking behind changing these (and others) from HashMap to ConcurrentHashMap when the accessors are only called from a single thread from the parser.

Reply:
Best practice is to avoid HashMap, even in the case where we are reasonably sure that we will only ever use one thread. The downside of the HashMap is too big, in case that some code is changed to be used in parallel. http://www.javaperformancetuning.com/news/news189.shtml